public/app/features/alerting/unified/components/rules/CloneRule.tsx

```diff
@@ -1,29 +1,39 @@
-import { css } from '@emotion/css';
 import React, { useState } from 'react';
-import { Redirect } from 'react-router-dom';
+import { Redirect, useLocation } from 'react-router-dom';
 
-import { GrafanaTheme2 } from '@grafana/data';
-import { Button, ConfirmModal, useStyles2 } from '@grafana/ui';
+import { Button, ConfirmModal } from '@grafana/ui';
 import { RuleIdentifier } from 'app/types/unified-alerting';
 
 import * as ruleId from '../../utils/rule-id';
 
 interface ConfirmCloneRuleModalProps {
   identifier: RuleIdentifier;
   isProvisioned: boolean;
+  redirectTo?: boolean;
```

Comment:
I might revert theses changes; we'll probably need a different approach here.

Currently when we include a `returnTo` we use that one when we `cancel` or `save`.

In this case we actually want `cancel` to go back to the rule but `save` to go to the new one.

Ideas? Should we split up cancel / save and add some weird additional complexity?

Reply:
I think we should split them. I wonder though in which case `Cancel` needs a `redirectTo` prop - Isn't it just going back in the history?

Reply:
Just took a stab at this and it's more complicated than I thought for some scenarios (mostly when going directly to the edit page itself).

This scenario would make users get stuck (but maybe we don't care?)

1. Go directly to the edit page of a rule
2. Click cancel –> there's no previous navigation so we can redirect the user to `/alerting/list` as a fallback.
3. Go forward in history to return to the edit page
4. Now we're in a state where we don't have a previous page (it doesn't go back to the "forward" page) but have a history stack > 1